src/output.ts

```diff
@@ -0,0 +1,59 @@
+import * as Serverless from "serverless";
+
+const yellowFont = "\x1b[33m";
+const underlineFont = "\x1b[4m";
+const endFont = "\x1b[0m";
+const outputPrefix = "DatadogMonitor";
+
+/**
+ * Builds the CloudFormation Outputs containing the alphanumeric key, description,
+ * and value (URL) to the function in Datadog
+ */
+export async function addOutputLinks(serverless: Serverless, site: string) {
+  const awsAccount = await serverless.getProvider("aws").getAccountId();
+  const region = serverless.service.provider.region;
+  const outputs = serverless.service.provider.compiledCloudFormationTemplate.Outputs;
+  if (outputs === undefined) {
+    return;
+  }
+
+  serverless.service.getAllFunctions().forEach((functionKey) => {
+    const functionName = serverless.service.getFunction(functionKey).name;
+    const key = `${outputPrefix}${functionKey}`.replace(/[^a-z0-9]/gi, "");
+    outputs[key] = {
+      Description: `See ${functionKey} in Datadog`,
+      Value: `https://app.${site}/functions/${functionName}:${region}:${awsAccount}:aws?source=sls-plugin`,
+    };
+  });
+}
+
+export async function printOutputs(serverless: Serverless) {
+  const stackName = `${serverless.service.getServiceName()}-${serverless.getProvider("aws").getStage()}`;
+  const describeStackOutput = await serverless
+    .getProvider("aws")
+    .request(
+      "CloudFormation",
+      "describeStacks",
```

Comment:
It's might be rare, but do you need to deal with the case where this API call fails (e.g., lack of permission) gracefully? e.g., logging a warning message (or even fail silently) instead of causing an exception to be displayed?

Reply:
I'm not too sure when it would fail if everything else was successful and it got to this `after:deploy:deploy` stage, however, just to be safe I added exception handling in the case it does. If it comes up it will fail silently and just skip the whole Datadog Monitoring section.